packages/noya-state/src/reducers/layerReducer.ts

```diff
@@ -195,6 +182,136 @@ export function layerReducer(
+        ids,
+      )
+        .filter((indexPath) =>
+          Layers.isChildLayer(Layers.access(page, indexPath)),
+        )
+        // Reverse the indexPaths to simplify deletion
+        .reverse();
+
+      const boundingRect = getBoundingRect(
+        page,
+        AffineTransform.identity,
+        ids,
+        { clickThroughGroups: true, includeHiddenLayers: true },
+      );
+
+      if (!boundingRect) {
+        console.info('[groupLayer] Selected layers not found');
+        return state;
+      }
+
+      const lastIndexPath = selectedIndexPaths[selectedIndexPaths.length - 1];
+      const newParentTransform = getLayerTransformAtIndexPath(
+        page,
+        lastIndexPath,
+      );
+      const groupFrame = transformRect(
+        boundingRect,
+        newParentTransform.invert(),
+      );
+      const newGroupTransform = AffineTransform.multiply(
+        newParentTransform,
+        AffineTransform.translation(groupFrame.x, groupFrame.y),
+      );
+
+      const symbolMasters = produce(Models.symbolMaster, (draft) => {
+        draft.do_objectID = uuid();
+        draft.name = name;
+        draft.frame = {
+          _class: 'rect',
+          constrainProportions: false,
+          ...groupFrame,
+        };
+        draft.style = produce(Models.style, (s) => {
+          s.do_objectID = uuid();
+        });
+        draft.symbolID = uuid();
+        draft.layers = [...selectedIndexPaths].reverse().map((indexPath) => {
+          const layer = Layers.access(page, indexPath) as Layers.ChildLayer;
+
+          const originalParentTransform = getLayerTransformAtIndexPath(
+            page,
+            indexPath,
+          );
+
+          // First we undo original parent's transform, then we apply the new group's transform
+          const transform = AffineTransform.multiply(
+            originalParentTransform,
+            newGroupTransform.invert(),
+          );
+
+          return produce(layer, (draftLayer) => {
+            draftLayer.frame = {
+              ...draftLayer.frame,
+              ...transformRect(draftLayer.frame, transform),
+            };
+          });
+        });
+      });
+
+      const symbolInstance = produce(Models.symbolInstance, (draft) => {
+        draft.do_objectID = uuid();
+        draft.name = name;
+        draft.frame = symbolMasters.frame;
+        draft.style = produce(Models.style, (s) => {
+          s.do_objectID = uuid();
+        });
+        draft.symbolID = symbolMasters.symbolID;
+      });
+
+      const createSymbolsPage = (
+        pages: Sketch.Page[],
+        user: Sketch.User,
+      ): Sketch.Page => {
+        const newPage = produce(Models.page, (page) => {
+          page.do_objectID = uuid();
+          page.name = 'Symbols';
+          return page;
+        });
+
+        user[newPage.do_objectID] = {
+          scrollOrigin: '{0, 0}',
+          zoomValue: 1,
+        };
+
+        pages.push(newPage);
+        return pages[pages.length - 2];
```

Comment:
Should this be `pages[pages.length - 1]`? It looks like this is returning the page _before_ the symbols page

Reply:
oh yess! I was doing something with it did notice I left it